modules/routes/Forums/forum_route.py

```diff
@@ -0,0 +1,91 @@
+from flask import request, jsonify
+from app import app
+from modules.models.account import AccountType
+from modules.routes.Forums.forum_schema import (
+    ForumSchema,
+    DiscussionThreadSchema,
+    DiscussionReplySchema,
+    NewDiscussionThreadSchema,
+    NewDiscussionReplySchema,
+
+)
+from modules.utils.db import db
+from modules.utils.route_utils import authenticate, fetch_session, protected_route
+
+@app.route("/forums/course/<string:course_code>", methods=["GET"])
+def get_course_forums(course_code):
```

Comment:
This method needs additional logic to check if the user may be a student/lecturer without access to the course. If that is the case the user should not be able to get the forum.

Also, rename and update it to account for the additional logic to be added.

```suggestion
def handle_couse_forums(course_code):
    if request.method == "GET":
        # Perform original get logic
    else:
        return create_course_forum(course_code)
```